---
title: "Manuscript Figures"
author: "Jackie Emathinger & Jonathan Nelson"
date: "2024-01-15"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Load libraries & Seurat Object

```{r load new packages, echo=FALSE, warning=FALSE, error=FALSE, message=FALSE}

if (!require("knitr")) {install.packages("knitr"); require("knitr")}
if (!requireNamespace('BiocManager', quietly = TRUE)) {install.packages('BiocManager'); require("BiocManager")}
if (!require("dplyr")) {install.packages("dplyr"); require("dplyr")}
if (!require("tidyverse")) {install.packages("tidyverse"); require("tidyverse")}


if (!require("clusterProfiler")) {BiocManager::install('clusterProfiler'); require("clusterProfiler")}
if (!require("enrichplot")) {BiocManager::install('enrichplot'); require("enrichplot")}


if (!require("ggrepel")) {install.packages("ggrepel"); require("ggrepel")}
if (!require("gghighlight")) {install.packages("gghighlight"); require("gghighlight")}
if (!require("ggpmisc")) {install.packages("ggpmisc"); require("ggpmisc")}
if (!require("ggupset")) {install.packages("ggupset"); require("ggupset")}
if (!require("ggplot2")) {install.packages("ggplot2"); require("ggplot2")}
if (!require("RColorBrewer")) {install.packages("RColorBrewer"); require("RColorBrewer")}

if (!require("AnnotationHub")) {BiocManager::install('AnnotationHub'); require("AnnotationHub")}
if (!require("org.Mm.eg.db")) {BiocManager::install('org.Mm.eg.db'); require("org.Mm.eg.db")}
if (!require("stringr")) {install.packages("stringr"); require("stringr")}
if (!require("car")) {install.packages("car"); require("car")}
if (!require("openxlsx")) {install.packages("openxlsx"); require("openxlsx")}
if (!require("readxl")) {install.packages("readxl"); require("readxl")}
if (!require("data.table")) {install.packages("data.table"); require("data.table")}
if (!require("here")) {install.packages("here"); require("here")}
if (!require("stats")) {install.packages("stats"); require("stats")}

library(stats)

#install.packages("kableExtra") # Install the package
library(kableExtra)           # Load the package

#install.packages("scCustomize")
library(scCustomize)

if(!require("magick")) {install.packages("magick",repos = "http://cran.us.r-project.org"); require("magick")}
if (!require("EnhancedVolcano")) {BiocManager::install('EnhancedVolcano'); require("EnhancedVolcano")} # volcano plot


library("Seurat")
library("ggplot2")
library("cowplot")

library("stringr")
here()

```
# Setup

To run this script to create the manuscript images you need the following files in the same folder that this RMD file is located in so that the here() function works:

1) The Seurat object "GSE253448_PTACE2KO.rds" from GSE253448
2) "ConSal_v_KOSal_DEGs.RData" and "ConAngII_v_KOAngII_DEGs.RData" which would be generated by running the "PT_ACE2_KO_AllSamples_clustering_jwn101723_DEGs.Rmd" script. 
3) "Comp1posgo.rds", "Comp1neggo.rds", "Comp2posgo.rds", and "Comp2neggo.rds" which would be generated by running the "PT_ACE2_KO_Transporters_11202023_jwnPTpathwaysb_jme.Rmd" script.

## Figure 7: Differential transcriptional responses to AngII in PT ACE2 KO and Control mice

### Figure 7A: Groups, replicates, and nuclei numbers in this dataset
```{r Figure7A,  echo=TRUE, warning=FALSE, error=FALSE, message=FALSE}

SO <- readRDS(here("GSE253448_PTACE2KO.rds"))

## extract meta data
md <- SO@meta.data %>% as.data.table
# the resulting md object has one "row" per cell

## count the number of nuclei per unique combinations of "Sample_ID" and "Experimental Groups"
md[, .N, by = c("Sample_ID", "Experimental_Groups")]

```

### Figure 7B: UMAP projection

```{r Figure 7B, echo=TRUE, warning=FALSE, error=FALSE, message=FALSE, fig.align = 'center', fig.height=5, fig.width=8}

Idents(SO)<-"class.jwn2"
f7B <- DimPlot(SO,
              reduction = "umap",
              label = FALSE,
              pt.size = 0,
              label.size = 6)
#check batch effects by treatment 

f7B

```

### Figure 7C: Dot plot displaying gene expression patterns of cluster-enriched markers

```{r Figure 7C, echo=TRUE, warning=FALSE, error=FALSE, message=FALSE, fig.align = 'center', fig.height=5, fig.width=6}

cluster_markers <- c(
                      "Slc5a12",      # PT-S1
                      "Slc13a3",      # PT-S2
                      "Slc16a9",      # PT-S3
                      "Epha7",        # dTL
                      "Slc12a1",      # TAL
                      "Nos1",         # MD
                      "Slc12a3",      # DCT
                      "Pvalb",        # DCT1
                      "Slc8a1",       # DCT2
                      "Aqp2",         # PC
                      "Slc4a1",       # IC-A
                      "Slc26a4",      # IC-B
                      "Nphs1",        # Podo
                      "Ncam1",        # PEC
                      "Flt1",         # Endo
                      "Pdgfrb",       # Perivascular
                      "Pdgfra",       # Fib
                      "Acta2",        # Mural
                      "Piezo2",       # Mesangial
                      "Cd74",         # Macrophage
                      "Skap1",        # B/T Cells 
                      "Upk1b"         # Uro
                      )


 f7C <- DotPlot(SO,
                features = cluster_markers,
        dot.scale = 8,
        dot.min = 0,
        scale.max = 100,
        scale.min = 0,
        col.min = -2.5,
        col.max = 2.5) +  
  theme(legend.position = "none", 
              plot.title = element_text(hjust = 0.5),
              axis.text.x = element_text(angle = 45, hjust = 1),
              axis.title.x = element_blank()) + 
  coord_flip()
 
 f7C
```


### Figure 7D: DEG Plots

```{r Figure 7D, echo=TRUE, warning=FALSE, error=FALSE, message=FALSE, fig.align = 'center', fig.height=5, fig.width=8}

# Control Saline vs. PT ACE2 KO DEGs

# ConSal_v_KOSal_DEGs.RData needs to be in the folder that contains this script

rm(list = ls())

load(here("ConSal_v_KOSal_DEGs.RData"))

# Remove the "_B" suffix from each of the objects in the environment

# List the objects in the environment
objects_in_environment <- ls()

# Filter the objects with the "_B" suffix
objects_to_rename <- grep("_B$", objects_in_environment, value = TRUE)

# Rename each object by removing the "_B" suffix
for (object in objects_to_rename) {
  new_name <- sub("_B$", "", object)
  assign(new_name, get(object))
  rm(list = object)
}

# Clean up by removing objects_to_rename
rm(objects_to_rename)


data_frames <- Filter(is.data.frame, mget(ls()))

# Initialize an empty data frame to store the results
output_table <- data.frame(Data_Frame = character(),
                           Number_of_Obs = numeric(),
                           Number_of_Positive_Log2FC = numeric(),
                           Number_of_Negative_Log2FC = numeric(),
                           stringsAsFactors = FALSE)

# Loop through each data frame
for (df_name in names(data_frames)) {
  df <- data_frames[[df_name]]
  
  # Get the number of observations in the data frame
  num_obs <- nrow(df)
  
  # Count the number of observations with positive log2FC
  positive_count <- sum(df$avg_log2FC > 0)
  
  # Count the number of observations with negative log2FC
  negative_count <- sum(df$avg_log2FC < 0)
  
  # Create a row in the output table
  output_row <- data.frame(Data_Frame = df_name,
                           Number_of_Obs = num_obs,
                           Number_of_Positive_Log2FC = positive_count,
                           Number_of_Negative_Log2FC = negative_count,
                           stringsAsFactors = FALSE)
  
  # Append the row to the output table
  output_table <- rbind(output_table, output_row)
}

# Print the output table
print(output_table)

# Measure the total number of DEGs (sum of output_table$Number_of_Obs)
v2 <-sum(output_table$Number_of_Obs)
print("Total Number of DEGs")
print(v2)

output_table2 <- output_table %>% 
                 mutate(Data_Frame = sub(".*\\.", "", Data_Frame)) %>%
                 arrange(desc(Number_of_Obs)) %>%
                 mutate(Data_Frame = factor(Data_Frame, levels = unique(Data_Frame)))

p2 <- ggplot(output_table2, aes(x = Data_Frame)) +
  geom_bar(aes(y = Number_of_Positive_Log2FC), stat = "identity", fill = "coral2") +
  geom_bar(aes(y = -Number_of_Negative_Log2FC), stat = "identity", fill = "dodgerblue2") +
  labs(x = "Cell Type", y = "Number of DEGs") +
  ggtitle("PT ACE2 KO Saline vs. Control Saline DEGs") +
  scale_y_continuous(labels = function(x) abs(x)) +
  annotate("text", x = Inf, y = Inf, label = "Positive Log2FC", color = "coral2",
           hjust = 1, vjust = 1, size = 8) +
  annotate("text", x = Inf, y = Inf, label = "Negative Log2FC", color = "dodgerblue2",
           hjust = 1, vjust = 2.1, size = 8) +
  theme_minimal() + 
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.title = element_text(hjust = 0.5, size = 24),
        axis.text = element_text(color = "black", size = 14,),
        axis.text.x = element_text(color = "black", size = 14, angle = 45, hjust = 1, vjust = 1),
        axis.title = element_text(color = "black", size = 20,))

p2

rm(list = ls())


# Control AngII vs. PT ACE2 KO AngII DEGs

# ConAngII_v_KOAngII_DEGs.RData needs to be in the folder that contains this script


rm(list = ls())

load(here("ConAngII_v_KOAngII_DEGs.RData"))

# Remove the "_D" suffix from each of the objects in the environment

# List the objects in the environment
objects_in_environment <- ls()

# Filter the objects with the "_D" suffix
objects_to_rename <- grep("_D$", objects_in_environment, value = TRUE)

# Rename each object by removing the "_D" suffix
for (object in objects_to_rename) {
  new_name <- sub("_D$", "", object)
  assign(new_name, get(object))
  rm(list = object)
}

# Clean up by removing objects_to_rename
rm(objects_to_rename)


data_frames <- Filter(is.data.frame, mget(ls()))

# Initialize an empty data frame to store the results
output_table <- data.frame(Data_Frame = character(),
                           Number_of_Obs = numeric(),
                           Number_of_Positive_Log2FC = numeric(),
                           Number_of_Negative_Log2FC = numeric(),
                           stringsAsFactors = FALSE)

# Loop through each data frame
for (df_name in names(data_frames)) {
  df <- data_frames[[df_name]]
  
  # Get the number of observations in the data frame
  num_obs <- nrow(df)
  
  # Count the number of observations with positive log2FC
  positive_count <- sum(df$avg_log2FC > 0)
  
  # Count the number of observations with negative log2FC
  negative_count <- sum(df$avg_log2FC < 0)
  
  # Create a row in the output table
  output_row <- data.frame(Data_Frame = df_name,
                           Number_of_Obs = num_obs,
                           Number_of_Positive_Log2FC = positive_count,
                           Number_of_Negative_Log2FC = negative_count,
                           stringsAsFactors = FALSE)
  
  # Append the row to the output table
  output_table <- rbind(output_table, output_row)
}

# Measure the total number of DEGs (sum of output_table$Number_of_Obs)
v4 <-sum(output_table$Number_of_Obs)
print("Total Number of DEGs")
print(v4)


# Print the output table
print(output_table)

output_table2 <- output_table %>% 
                 mutate(Data_Frame = sub(".*\\.", "", Data_Frame)) %>%
                 arrange(desc(Number_of_Obs)) %>%
                 mutate(Data_Frame = factor(Data_Frame, levels = unique(Data_Frame)))

p3 <- ggplot(output_table2, aes(x = Data_Frame)) +
  geom_bar(aes(y = Number_of_Positive_Log2FC), stat = "identity", fill = "coral2") +
  geom_bar(aes(y = -Number_of_Negative_Log2FC), stat = "identity", fill = "dodgerblue2") +
  labs(x = "Cell Type", y = "Number of DEGs") +
  ggtitle("PT ACE2 KO AngII vs. Control AngII DEGs") +
  scale_y_continuous(labels = function(x) abs(x)) +
  annotate("text", x = Inf, y = Inf, label = "Positive Log2FC", color = "coral2",
           hjust = 1, vjust = 1, size = 8) +
  annotate("text", x = Inf, y = Inf, label = "Negative Log2FC", color = "dodgerblue2",
           hjust = 1, vjust = 2.1, size = 8) +
  theme_minimal() + 
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        plot.title = element_text(hjust = 0.5, size = 24),
        axis.text = element_text(color = "black", size = 14,),
        axis.text.x = element_text(color = "black", size = 14, angle = 45, hjust = 1, vjust = 1),
        axis.title = element_text(color = "black", size = 20,))

p3

rm(list = ls())


```


### Figure 7E

```{r Figure 7E, echo=TRUE, warning=FALSE, error=FALSE, message=FALSE, results=T, fig.height = 6, fig.width = 10}

SO <- readRDS(here("GSE253448_PTACE2KO.rds"))

RAS<-c( "Mas1", "Cltrn", "Ace2", "Adam17","Agtr1a",  "Ace",  "Agt", "Ren1")

# subset out clusters that express RAS components
  #PTS1-3
  #Contractile cells (pericytes) express Ren1 and Agtr1a
  #Mesangial cells express Agtr1a
  #Macrophages express Adam17
Idents(SO)<-"class.jwn2"
RAS_idents <- subset(SO,  idents = c("PTS1", "PTS2", "PTS3", "Contractile", "Mes", "Macro"))


f7E <-DotPlot(RAS_idents,
  features = c(RAS),
  group.by = "Experimental_Groups",
  dot.scale = 8,
  dot.min = 0,
  scale.max = 20,
  scale.min = 0,
  scale.by ="size",
  col.min = -1,
cols = c("grey85", "blue"),
  col.max = 1) +
  theme(axis.text.x = element_text(angle = 0, hjust = 0.5)) + ggtitle("Intrarenal RAS Components") +
  theme(plot.title = element_text(hjust = 0.5)) +
  coord_flip() + 
  xlab('')+ ylab('')

f7E

```


## Figure 8: Enrichment of pathways found significant in PT ACE2 KO Saline vs. Control Saline groups
### Figure 8A Upregulated pathways
```{r Figure 8A-up, echo=TRUE, warning=FALSE, error=FALSE, message=FALSE, fig.align = 'center', fig.height=5, fig.width=6}
## subset only proximal tubule clusters
SO_PT <- subset(SO, idents = c("PTS1", "PTS2", "PTS3"))


# Upregulated pathways
Comp1pos_go<-readRDS(here("Comp1posgo.rds"))

# Unique GO terms:
  #proton transmembrane transporter activity GO:0015078
  #glomerular basement membrane development GO:0032836
  #transmembrane transporter complex GO:1902495
  #proton transmembrane transport GO:1902600
  #transporter complex GO:1990351

 
genes <- Comp1pos_go@result$geneID
 

#proton transmembrane transporter activity GO:0015078
go_term <- 'GO:0015078'
# Select the geneID column for the specified GO term
geneID <- Comp1pos_go$geneID[Comp1pos_go$ID == go_term]
# Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:0015078_")


#glomerular basement membrane development GO:0032836
go_term <- 'GO:0032836'
# Select the geneID column for the specified GO term
geneID <- Comp1pos_go$geneID[Comp1pos_go$ID == go_term]
# Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:0032836_")


#transmembrane transporter complex GO:1902495
go_term <- 'GO:1902495'
# Select the geneID column for the specified GO term
geneID <- Comp1pos_go$geneID[Comp1pos_go$ID == go_term]
# Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:1902495_")


#proton transmembrane transport GO:1902600
go_term <- 'GO:1902600'
# Select the geneID column for the specified GO term
geneID <- Comp1pos_go$geneID[Comp1pos_go$ID == go_term]
# Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:1902600_")


#transporter complex GO:1990351
go_term <- 'GO:1990351'
# Select the geneID column for the specified GO term
geneID <- Comp1pos_go$geneID[Comp1pos_go$ID == go_term]
# Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:1990351_")
```

#### Downregulated pathways
```{r Figure 8A-down, echo=TRUE, warning=FALSE, error=FALSE, message=FALSE, fig.align = 'center', fig.height=5, fig.width=6}
Comp1neg_go<-readRDS(here("Comp1neggo.rds"))
# Unique GO terms:
  #proline transport GO:0015824
  #response to sodium phosphate GO:1904383
 #negative regulation of transmembrane transport GO:0034763
 # cellular response to metal ion GO:0071248
  #negative regulation of transforming growth factor beta receptor signaling pathway GO:0030512

genes <- Comp1neg_go@result$geneID

#proline transport GO:0015824 
go_term <- 'GO:0015824'
# Select the geneID column for the specified GO term
geneID <- Comp1neg_go$geneID[Comp1neg_go$ID == go_term]
# Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:0015824_")



#response to sodium phosphate GO:1904383
go_term <- 'GO:1904383'
# Select the geneID column for the specified GO term
geneID <- Comp1neg_go$geneID[Comp1neg_go$ID == go_term]
# Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:1904383_")


#negative regulation of transmembrane transport GO:0034763
go_term <- 'GO:0034763'
# Select the geneID column for the specified GO term
geneID <- Comp1neg_go$geneID[Comp1neg_go$ID == go_term]
# Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:0034763_")


# cellular response to metal ion GO:0071248
go_term <- 'GO:0071248'
# Select the geneID column for the specified GO term
geneID <- Comp1neg_go$geneID[Comp1neg_go$ID == go_term]
# Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:0071248_")


#negative regulation of transforming growth factor beta receptor signaling pathway GO:0030512
go_term <- 'GO:0030512'
# Select the geneID column for the specified GO term
geneID <- Comp1neg_go$geneID[Comp1neg_go$ID == go_term]
# Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:0030512_")

head(SO_PT@meta.data)

```
### Figure 8A
```{r Figure8A, echo=TRUE, warning=FALSE, error=FALSE, message=FALSE, fig.align = 'center', fig.height=5, fig.width=9}
KO_Saline_v_Ctrl_Saline<- DotPlot(SO_PT,
  features = c( "GO:0015824_1", "GO:1904383_1", "GO:0034763_1", "GO:0071248_1", "GO:0030512_1", "GO:0015078_1", "GO:0032836_1", "GO:1902495_1", "GO:1902600_1", "GO:1990351_1"),
  group.by = "Experimental_Groups",
  dot.scale = 8,
  dot.min = 0,
  scale.max = 100,
  scale.min = 0,
  col.min = -2.5,
    cols = c("dodgerblue", "firebrick1"), 
  col.max = 2.5) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ggtitle("PT ACE2 KO Saline vs. Control Saline") +
  theme(plot.title = element_text(hjust = 0.5)) +
  coord_flip() +
  scale_x_discrete(labels = c( "proline transport", "response to sodium phosphate", "negative regulation of transmembrane transport", "cellular response to metal ion", "negative regulation of TGFbR signaling pathway", "proton transmembrane transporter activity", "glomerular basement membrane development", "transmembrane transporter complex", "proton transmembrane transport", "transporter complex")) + 
  xlab('')+ ylab('')

    
print(KO_Saline_v_Ctrl_Saline)
```

## Figure 8B Enrichment of pathways in PT ACE2 KO AngII vs. Control AngII
### Upregulated pathways
```{r Figure 8B-up, echo=TRUE, warning=FALSE, error=FALSE, message=FALSE, fig.align = 'center', fig.height=5, fig.width=6}
Comp2pos_go<-readRDS(here("Comp2posgo.rds"))
# Unique GO terms:
  #sodium ion transmembrane transporter activity GO:0015081
  #monoatomic anion:monoatomic cation symporter activity GO:0015296
  #sodium-independent organic anion transmembrane transporter activity GO:0015347
  #solute:sodium symporter activity GO:0015370
  #positive regulation of ion transmembrane transporter activity GO:0032414

genes <- Comp2pos_go@result$geneID

# sodium ion transmembrane transporter activity GO:0015081
go_term <- 'GO:0015081'
# Select the geneID column for the specified GO term
geneID <- Comp2pos_go$geneID[Comp2pos_go$ID == go_term]
 # Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:0015081_")


#monoatomic anion:monoatomic cation symporter activity GO:0015296
go_term <- 'GO:0015296'
# Select the geneID column for the specified GO term
geneID <- Comp2pos_go$geneID[Comp2pos_go$ID == go_term]
 # Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
 
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:0015296_")


#sodium-independent organic anion transmembrane transporter activity GO:0015347
go_term <- 'GO:0015347'
# Select the geneID column for the specified GO term
geneID <- Comp2pos_go$geneID[Comp2pos_go$ID == go_term]
 # Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
 
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:0015347_")


#solute:sodium symporter activity GO:0015370
go_term <- 'GO:0015370'
# Select the geneID column for the specified GO term
geneID <- Comp2pos_go$geneID[Comp2pos_go$ID == go_term]
 # Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
 
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:0015370_")

 
#positive regulation of ion transmembrane transporter activity GO:0032414
go_term <- 'GO:0032414'
# Select the geneID column for the specified GO term
geneID <- Comp2pos_go$geneID[Comp2pos_go$ID == go_term]
 # Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
 
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:0032414_")

head(SO_PT@meta.data)

```

### Downregulated pathways
```{r Figure 8B-down, echo=TRUE, warning=FALSE, error=FALSE, message=FALSE, fig.align = 'center', fig.height=5, fig.width=6}
Comp2neg_go<-readRDS(here("Comp2neggo.rds"))
# Unique GO terms
  # regulation of amino acid import across plasma membrane GO:0010958
  #neutral L-amino acid transmembrane transporter activity GO:0015175
  #regulation of transmembrane transporter activity GO:0022898
  #renal absorption GO:0070293
  # sodium ion transmembrane transport GO:0035725

genes <- Comp2neg_go@result$geneID
 
# regulation of amino acid import across plasma membrane GO:0010958
go_term <- 'GO:0010958'
# Select the geneID column for the specified GO term
geneID <- Comp2neg_go$geneID[Comp2neg_go$ID == go_term]
# Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:0010958_")


#neutral L-amino acid transmembrane transporter activity GO:0015175
go_term <- 'GO:0015175'
# Select the geneID column for the specified GO term
geneID <- Comp2neg_go$geneID[Comp2neg_go$ID == go_term]
# Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:0015175_")


#regulation of transmembrane transporter activity GO:0022898
go_term <- 'GO:0022898'
# Select the geneID column for the specified GO term
geneID <- Comp2neg_go$geneID[Comp2neg_go$ID == go_term]
# Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:0022898_")

#renal absorption GO:0070293
go_term <- 'GO:0070293'
# Select the geneID column for the specified GO term
geneID <- Comp2neg_go$geneID[Comp2neg_go$ID == go_term]
# Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:0070293_")

# sodium ion transmembrane transport GO:0035725
go_term <- 'GO:0035725'
# Select the geneID column for the specified GO term
geneID <- Comp2neg_go$geneID[Comp2neg_go$ID == go_term]
# Print or use the selected GeneIDs
print(geneID)
geneID <- unlist(geneID)
geneID <- unlist(strsplit(geneID, "/"))
geneID <- list(geneID)
print(geneID)
SO_PT <- AddModuleScore(object = SO_PT, features = geneID, name = "GO:0035725_")

```

### Figure 8B
```{r Figure 8B, echo=TRUE, warning=FALSE, error=FALSE, message=FALSE, fig.align = 'center', fig.height=5, fig.width=10}
KO_AngII_v_Ctrl_AngII<- DotPlot(SO_PT,
  features = c("GO:0010958_1", "GO:0015175_1",  "GO:0022898_1", "GO:0070293_1", "GO:0035725_1",
    
    "GO:0015081_1", "GO:0015296_1", "GO:0015347_1", "GO:0015370_1", "GO:0032414_1"),
  group.by = "Experimental_Groups",
  dot.scale = 8,
  dot.min = 0,
  scale.max = 100,
  scale.min = 0,
  col.min = -2.5,
    cols = c("dodgerblue", "firebrick1"), 
  col.max = 2.5) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + ggtitle("PT ACE2 KO AngII vs. Control AngII") +
  theme(plot.title = element_text(hjust = 0.5)) +
  coord_flip() +
  scale_x_discrete(labels = c(  "regulation of amino acid import across plasma membrane", "neutral L-amino acid transmembrane transporter activity",  "regulation of transmembrane transporter activity", "renal absorption",  "sodium ion transmembrane transport", # down in KOAngII v Ctrl AngII 
                               
                 "sodium ion transmembrane transporter activity", "monoatomic anion:monoatomic cation symporter activity","Na+ -ind. organic anion transmembrane transporter activity", "solute:sodium symporter activity", "positive regulation of ion transmembrane transporter activity" # Up KOAngII v Ctrl AngII
                 ))+ 
  xlab('')+ ylab('')

print(KO_AngII_v_Ctrl_AngII)
```